cobc/ChangeLog

```diff
@@ -1,5 +1,28 @@
 
-2023-01-23 David Declerck <david.declerck@ocamlpro.com>
+2023-01-28  Fabrice Le Fessant <fabrice.le_fessant@ocamlpro.com>
+
+	* cobc.c (process_command_line): disable auto-detection of format
+	  if -f or -F options are provided.
+	* cobc.c (main): initialize cb_config_text_column to 72 to avoid a
+	  race condition in config files.
+	* cobc.c (preprocess): try to autodetect the format of a file
```

Comment:
As mentioned before: please move that call to ppopen, this way it is done one each "main source" again, and we cover the BOM. We can also get rid of the extern autoformat, by just setting the default format to "unknown" and do the check in ppopen if the format is "still unknown".

Reply:
What do you mean by BOM ?

`ppopen` is also used to read copybooks, so I will have to change its interface to add an additional parameter to tell it whether autodetection should be performed or not.

Reply:
> What do you mean by BOM ?

The byte-order-mark which ppopen already handles.

> `ppopen` is also used to read copybooks, so I will have to change its interface to add an additional parameter to tell it whether autodetection should be performed or not.

I _think_ it should be able to just check "is the format still undefined" - if yes then we are at a main program.
If it isn't, then it likely should be that way ;-) Otherwise `copy_stack == NULL` would do the trick, but I'd prefer the former.

Reply:
done.